File: FocusLit-Server/src/main/java/com/focus/lit/repository/UserRepository.java
package com.focus.lit.repository;

import com.focus.lit.model.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Integer> {
    Optional<User> findById(int id);
    Optional<User> findByMail(String mail);
    Optional<User> findByEmailVerificationLink(String emailVerificationLink);

}

File: FocusLit-Android/app/src/main/java/com/focus/lit/data/repository/UserRepository.kt
package com.focus.lit.data.repository

import com.focus.lit.data.model.UserInfo
import com.focus.lit.data.remote.ApiService

class UserRepository(private val apiService: ApiService) {
    suspend fun getUser(userId: Int): UserInfo {
        return apiService.getUser(userId)
    }
}
